ci/tests/test_01_functional/test_03_scanner/__init__.py

```diff
@@ -0,0 +1,185 @@
+        # project name generated from appid-jobid-tag
+        self.project_under_test = BUILD_FAIL_PROJECT_NAME
+        # initialize projects model, to simulate cccp-index job
+        self.appid = "nshaikh"
+        self.jobid = "go-helloworld"
+        self.desired_tag = "latest"
+        self.repo_url = "https://github.com/navidshaikh/minimal-go-container"
+        self.repo_branch = "master"
+        self.repo_build_path = "/"
+        self.target_file = "Dockerfile.scratch"
+        self.depends_on = "centos/centos:latest"
+        self.test_tag = "SCANNERS_TEST"
+        self.logs_dir = "/srv/pipeline-logs/" + self.test_tag
+        self.build_context = "./"
+        self.build_number = str(randint(11, 99))
+        self.notify_email = "container-status-report@centos.org"
+        self.cleanup_beanstalkd()
+        self.cleanup_openshift()
+
+    def run_dj_script(self, script):
+        _script = (
+            'import os, django; '
+            'os.environ.setdefault(\\"DJANGO_SETTINGS_MODULE\\", '
+            '\\"container_pipeline.lib.settings\\"); '
+            'django.setup(); '
+            '{}'
+        ).format(script)
+        return self.run_cmd(
+            'cd /opt/cccp-service && '
+            'python -c "{}"'.format(_script))
+
+    def start_build(self):
+        """
+        Starts the build of a project
+        """
+        self.provision()
+        # create database entry for Project model for project under test
+        self.run_dj_script(
+            'from container_pipeline.models import Project; '
+            'Project.objects.get_or_create('
+            'name=\\"nshaikh-go-helloworld-latest\\")'
+        )
+        workspace_dir = os.path.join(
+            "/srv/jenkins/workspace/",
+            self.project_under_test)
+
+        # run the container_pipeline/pipeline.py module
+        args = " ".join([
+            self.appid, self.jobid, self.repo_url,
+            self.repo_branch, self.repo_build_path,
+            self.target_file, self.notify_email,
+            self.desired_tag, self.depends_on,
+            self.test_tag, self.build_number,
+            self.build_context])
+
+        command = ("mkdir -p {0} && "
+                   "git clone {1} {0} && "
+                   "export DOCKERFILE_DIR={0} && "
+                   "export PYTHONPATH=/opt/cccp-service && "
+                   "mkdir -p /srv/pipeline-logs/{2} && "
+                   "cd /opt/cccp-service && "
+                   "python container_pipeline/pipeline.py {3}").format(
+            workspace_dir, self.repo_url, self.test_tag, args)
+
+        # we need to trigger build on jenkins_slave node
+        # figure out jenkins_slave hostname to run command upon
+        # self.hosts variable is defined in base.py
+        jenkins_slave = self.hosts["jenkins_slave"]["host"]
+
+        print "Starting a test build with command: \n%s" % command
+        print self.run_cmd(command, host=jenkins_slave)
```

Comment:
we are manually running the command in jenkins slave. So we do not need the jenkins job entry for this.